### PR TITLE
Enable extreme comparisons lint

### PR DESCRIPTION
**Pull request by shssoichiro**
https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons

### CODE REVIEW

**Comment by lu-zero** [comment]
This lint feels wrong to me. @tdaede am I alone in that? :)

**Comment by shssoichiro** [comment]
Why is that? If the variable is a `usize`, it can't ever be less than 0. Unless the concern is that it might change to an `isize` in the future?

File: src/context.rs [hunk 1]
```diff
@@ -2559,9 +2559,6 @@ impl ContextWriter {
       /* TODO: Set zeromv ref to the converted global motion vector */
     } else {
       /* TODO: Set the zeromv ref to 0 */
-    }
-
-    if ref_frames[0] <= INTRA_FRAME {
```
Comment (tdaede) [inline src/context.rs]: Would be interesting to see why this condition was there.

File: src/deblock.rs [hunk 2]
```diff
@@ -121,8 +121,8 @@ fn deblock_size<T: Pixel>(
   if !(block_edge
     || !block.skip
     || !prev_block.skip
-    || block.ref_frames[0] <= INTRA_FRAME
-    || prev_block.ref_frames[0] <= INTRA_FRAME)
+    || block.ref_frames[0] == INTRA_FRAME
```
Comment (tdaede) [inline src/deblock.rs]: This change is actually clearer.

File: src/transform/mod.rs [hunk 3]
```diff
@@ -37,7 +37,7 @@ fn half_btf(w0: i32, in0: i32, w1: i32, in1: i32, bit: usize) -> i32 {
   //   overflows, but w0*in0 + w1*in1 + rounding isn't.
   let result = (w0 * in0).wrapping_add(w1 * in1);
   // Implement a version of round_shift with wrapping
-  if bit <= 0 {
+  if bit == 0 {
```
Comment (tdaede) [inline src/transform/mod.rs]: meh on this one

**Comment by tdaede** [comment]
I would say this is a less useful lint, but it did find some rather odd code.

File: src/context.rs (hunk 1 above)
Comment (tdaede) [reply src/context.rs]: Ohh, I see, it's collapsing this condition into the one above. That's fancy.

**Pull request merged**